Add more feature docs (#848)
rancher/rio: docs/install.md

If you have limited capacities in your cluster and don't want enable some of features, you can disable features to make footprint of Rio smaller.

Here is the list of features:

| Feature     | Description                                       |
|-------------|---------------------------------------------------|
| autoscaling | Auto-scaling services based on in-flight requests |
| build       | Rio Build, from source code to deployment         |
| gloo        | API gateway backed by gloo                        |
| linkerd     | Linkerd service mesh                              |
| letsencrypt | Let's Encrypt                                     |
| rdns        | Acquire DNS from public Rancher DNS service       |
| dashboard   | Rio UI                                            |

The core functionality of Rio includes Rio CLI UX and Riofile UX, which is built-in and doesn't rely on any of these features. Beside core functionality, here is a list of advanced features.

**Autoscaling**

> This is the feature that manages autoscaling feature in Rio. It deploys a pod inside cluster, watch autoscaling config for each service and scrape metrics from pods. According to real time metrics and autoscaling config, it will make decisions to scale pods up and down.

Reviewer: ```suggestion
Manages autoscaling services in Rio. It deploys a pod inside cluster, watches the autoscaling config for each service, and scrapes metrics from pods. It will then make decisions to scale pods up and down based on those real-time metrics and autoscaling configs.
```

**Build**

Build is responsible for building images from source code, which allows you to run deployment directly from source code with a Dockerfile. Build is also responsible for setting up webhook to watch your source code change and trigger automatic builds.

**Gloo**

Gloo is the default API gateway Rio deploys to manage ingress traffic outside of cluster. It uses [Envoy](https://www.envoyproxy.io/) under the hood to provide rate limiting, circuit breaking, retries and other advanced routing features. Rio router programs Gloo virtualservices.

**Linkerd**

> Linkerd is a service mesh for Kubernetes and other frameworks. It makes running services easier and safer by giving you runtime debugging, observability, reliability, and security—all without requiring any changes to your code. Rio by default ships Linkerd and enable service mesh within Rio services.

Reviewer: ```suggestion
Linkerd is a service mesh for Kubernetes and other frameworks. It makes running services easier and safer by giving you runtime debugging, observability, reliability, and security—all without requiring any changes to your code. By default Rio ships with Linkerd and enables its service mesh within Rio services.
```

**Letsencrypt**

> Letsencrypt is default CA that Rio uses to provision and sign certificates and use that to provide TLS traffic for your workloads. You easily swap certificates to your own certs. Under the hood it deploys [cert-manager](https://github.com/jetstack/cert-manager) to manage certificate provisioning.

Reviewer: You should link here to your cert code below.

**Rdns**

Rdns is public DNS service hosted by Rancher. Each cluster with Rio installed will get a unique cluster domain (xxxxxx.on-rio.io). Rio registers this domain to Loadbalancer IP of API gateway service, which will eventually point to your service. Rdns will keep DNS record up-to-date if loadbalancer IP has changed.

> If you disabled this feature, make sure you have ways to update your own dns record if loadbalancer IP has changed.

Reviewer: ```suggestion
If you disabled this feature, you are responsible for updating your own DNS record if the loadbalancer IP changes.
```

**Dashboard**

> Dashboard is the built-in dashboard for Rio.

Reviewer: ```suggestion
Dashboard is a built-in web UI for Rio that runs within the cluster. It can be used as an alternative to the Rio CLI. 
```
